cpp_code/tests/test_random_number_generator.cpp

```diff
@@ -28,6 +28,36 @@ using std::vector;
 
 static const size_t NSAMPLES = 100000;
 
+// The tests are calibrated to have significance level alpha = 0.01,
+// i.e. the rejection rate under the null hypothesis, or false
+// rejection rate.  If there are n tests, then the false rejection
+// rate of the entire suite is 1 - Binom(0; n, alpha).  For n = 10,
+// this is about 10%, i.e. about one in ten runs of the test suite
+// would fail even if the code is correct.
+//
+// To reduce the false rejection rate, we could change alpha, but that
+// requires changing all the critical values in all the tests.
+// Critical values for smaller values of alpha are not always easy to
+// find and audit, e.g. the critical value for the smallest alpha
+// reported in the Shapiro-Wilk paper is 0.01.
+//
+// An easy alternative way to reduce the false rejection rate, at the
+// expense of statistical power, is to repeat each test in k
+// independent trials and reject only if it failed all k.  It is then
+// as if the significance level were Binom(k; k, alpha).  For k = 2,
+// this is approximately alpha^2.  For alpha = 0.01, the resulting
+// false rejection rate of the entire suite for n = 10 tests is then
+// approximately 1 - Binom(0; 10, alpha^2) or about 0.1%.  If we wrote
+// 90 more tests, then the false rejection rate would still be only
+// about 1% for the whole test suite.
+//
+// We could also stop early as soon as we cross the threshold of
+// required passes, without changing the false rejection rate, but
+// that requires writing more code.
+
+static const unsigned NPASSES = 1;
```

Comment:
I intended the "Consider", the "?" and the top-level "LGTM" to indicate that you should do what you feel is best, including nothing at all.